pkg/testutils/reduce/reducesql/reducesql.go

```diff
@@ -672,14 +672,19 @@ var (
 		return 0
 	})
 	removeIndexCols = walkSQL("remove INDEX cols", func(xfi int, node interface{}) int {
-		switch node := node.(type) {
-		case *tree.IndexTableDef:
-			n := len(node.Columns)
-			if xfi < len(node.Columns) {
-				node.Columns = append(node.Columns[:xfi], node.Columns[xfi+1:]...)
+		removeCol := func(idx *tree.IndexTableDef) int {
+			n := len(idx.Columns)
+			if xfi < len(idx.Columns) {
+				idx.Columns = append(idx.Columns[:xfi], idx.Columns[xfi+1:]...)
 			}
 			return n
 		}
+		switch node := node.(type) {
+		case *tree.IndexTableDef:
+			return removeCol(node)
+		case *tree.UniqueConstraintTableDef:
```

Comment:
do you want to case on whether or not `PrimaryKey` is true, or just apply it to all `UniqueConstraintTableDef`s?

Reply:
All index types benefit from this reduction, so yes, it should apply to both here.